test/sol6/kyberNetwork.js

```diff
@@ -2180,6 +2182,137 @@ contract('KyberNetwork', function(accounts) {
+                }
+            });
+
+            after("clean up", async() => {
+                await network.removeKyberProxy(networkProxy, {from: admin});
+                await nwHelper.removeReservesFromStorage(storage, reserveInstances, tokens, operator);
+                reserveInstances = {};
+            });
+
+            it("should success when e2t with dgx, network not pays fee", async() => {
+                let ethSrcQty = new BN(10).pow(new BN(18));
+                let hintType = EMPTY_HINTTYPE;
+                hint = await nwHelper.getHint(rateHelper, matchingEngine, reserveInstances, hintType, undefined, ethAddress, dgxToken.address, ethSrcQty);
+
+                info = [ethSrcQty, networkFeeBps, platformFeeBps];
+
+                expectedResult = await nwHelper.getAndCalcRates(matchingEngine, storage, reserveInstances,
+                    ethAddress, dgxToken.address, ethSrcQty,
+                    ethDecimals, dgxDecimal,
+                    networkFeeBps, platformFeeBps, hint);
+
+                let initialReserveBalances = await nwHelper.getReserveBalances(ethAddress, dgxToken, expectedResult);
+                let initialTakerBalances = await nwHelper.getTakerBalances(ethAddress, dgxToken, taker, networkProxy);
+                let initialNetworkDgxBalance = await dgxToken.balanceOf(network.address);
+                [expectedResult, actualSrcQty] = await nwHelper.calcParamsFromMaxDestAmt(ethAddress, dgxToken, expectedResult, info, maxDestAmt);
+
+                await network.tradeWithHintAndFee(networkProxy, ethAddress, ethSrcQty, dgxToken.address, taker,
+                    maxDestAmt, minConversionRate, zeroAddress, platformFeeBps, hint, {value: ethSrcQty, from: networkProxy, gasPrice: new BN(0)});
+
+                await nwHelper.compareBalancesAfterTrade(ethAddress, dgxToken, actualSrcQty,
+                    initialReserveBalances, initialTakerBalances, expectedResult, taker, networkProxy);
+                
+                //because network is in whitelist so fee is not change
+                await Helper.assertSameTokenBalance(network.address, dgxToken, initialNetworkDgxBalance);
+            });
+
+            it("should success when t2e with dgx, network pays fee", async() => {
+                let srcQty = new BN(10).pow(new BN(9));
+                let hintType = EMPTY_HINTTYPE;
+                hint = await nwHelper.getHint(rateHelper, matchingEngine, reserveInstances, hintType, undefined, dgxToken.address, ethAddress, srcQty);
+
+                info = [srcQty, networkFeeBps, platformFeeBps];
+
+                expectedResult = await nwHelper.getAndCalcRates(matchingEngine, storage, reserveInstances,
+                    dgxToken.address, ethAddress, srcQty,
+                    dgxDecimal, ethDecimals,
+                    networkFeeBps, platformFeeBps, hint);
+                await dgxToken.mintDgxFor(networkProxy, srcQty, {from: admin});
```

Comment:
Oh
I understand proxy is the source address of the trade here. so its "taker"

seems to me better if adding a real proxy here 
(even though its network file.) we can add it since essentially we are testing the network.
And with proxy we test real flow.

Reply:
change as request